AmplifyTestCommon/AsyncExpectation.swift

```diff
@@ -27,6 +27,14 @@ public actor AsyncExpectation {
     public var isFulfilled: Bool {
         state == .fulfilled
     }
+    
+    public func setShouldTrigger(_ shouldTrigger: Bool) {
+        self.isInverted = !shouldTrigger
+    }
+    
+    public func setExpectedFulfillmentCount(_ count: Int) {
+        self.expectedFulfillmentCount = count
+    }
```

Comment:
CC @brennanMKE  I had a bunch of tests that created the expectations in the setUp, then in the test methods control the `isInverted` and `fulfillmentCount`, which were different across test methods. let me know if you have any comments for these changes